File: src/types/history.ts
import type { Metrics } from './metrics'

export type HistoryEntry = {
  id: string
  testedAt: string
  gameId: string
  gameName: string
  nodeId: string
  nodeName: string
  before?: Metrics
  after: Metrics
  result: 'improved' | 'worse' | 'stable'
}

File: src-tauri/src/models/history.rs
use serde::{Deserialize, Serialize};

use super::metrics::Metrics;

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct HistoryEntry {
    pub id: String,
    pub tested_at: String,
    pub game_id: String,
    pub game_name: String,
    pub node_id: String,
    pub node_name: String,
    pub before: Option<Metrics>,
    pub after: Metrics,
    pub result: String,
}
